### PR TITLE
[VCA-11] Game Night Support

### PR DESCRIPTION
**Pull request by val-edu-center**

### CODE REVIEW

File: src/components/common/CheckboxInput.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        <input type="checkbox" name={name} onChange={onChange}></input>
+        <label htmlFor={name}>{name}</label>
+    </div>
+}
+
+CheckboxInput.propTypes = {
+    name: PropTypes.string.isRequired,
+    label: PropTypes.string.isRequired,
+    onChange: PropTypes.func.isRequired,
+    placeholder: PropTypes.string,
```
Comment (jpcanty) [inline src/components/common/CheckboxInput.js]: remove

File: src/components/common/CheckboxInput.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import React from "react"
+import PropTypes from "prop-types"
+
+const CheckboxInput = ({ name, label, values, onChange, error }) => {
+    let wrapperClass = "form-group";
+    if (error && error.length > 0) {
+        wrapperClass += " " + "has-error";
+    }
+    const possibleOptions = ["Ya", "Yeet", "Mardi", "Gras", "Dat", "Boi"]
```
Comment (jpcanty) [inline src/components/common/CheckboxInput.js]: TODO: make this configurable

File: src/components/gameNights/GameNightPage.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            actions.gameNight.loadAllVotes().catch(error => {
+                alert("Loading game night votes failed " + error)
+            })
+        }
+    }
+
+    render() {
+        return (
+            <>
+                {!this.props.session.sessionToken && <Redirect to="/unauthorized" />}
```
Comment (jpcanty) [inline src/components/gameNights/GameNightPage.js]: add is admin check

File: src/components/gameNights/ManageGameNightPage.js [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+import Spinner from "../common/Spinner";
+import Parse from 'parse/dist/parse.min.js'
+import { toast } from "react-toastify";
+
+const ManageGameNightPage = ({ showSpinner, gameNights, actions, history, ...props }) => {
+    //This is how React Hooks add state to function components
+    const [gameNight, setGameNight] = useState({ ...props.gameNight })
+    const [errors, setErrors] = useState({ ...props.errors })
+    const [saving, setSaving] = useState(false)
+    console.log(gameNights)
```
Comment (jpcanty) [inline src/components/gameNights/ManageGameNightPage.js]: remove

File: src/components/gameNights/ManageGameNightPage.js [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        }
+        
+        //useEffect with an empty array is equivalent to componentDidMount
+        //Otherwise, would run everytime it renders
+    }, [props.gameNight])
+
+    function changeOptionList(gameNight, option, checked) {
+        const oldParseObject = gameNight.parseObject
+        const newOptions = checked ? [...gameNight.options, option] : gameNight.options.filter(o => o !== option)
+        if (oldParseObject) {
```
Comment (jpcanty) [inline src/components/gameNights/ManageGameNightPage.js]: remove this check and always set parse object

File: src/components/gameNights/ManageGameNightPage.js [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+        const option = event.target.name
+        const newGameNight = changeOptionList(gameNight, option, event.target.checked)
+        setGameNight(newGameNight)
+
+    }
+
+    function changeDate(gameNight, newDate) {
+        const oldParseObject = gameNight.parseObject
+        if (oldParseObject) {
+            oldParseObject.set("date", new Date(newDate))
```
Comment (jpcanty) [inline src/components/gameNights/ManageGameNightPage.js]: remove check

File: src/components/gameNights/ManageGameNightPage.js [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+    return {
+        gameNights: state.gameNight.dates,
+        gameNight,
+        showSpinner: showSpinner,
+        errors: []
+    }
+}
+
+function createNewGameNight() {
```
Comment (jpcanty87) [inline src/components/gameNights/ManageGameNightPage.js:141]: move upward

**Pull request merged**